File: 2020/day01/python/report_repair.py
with open("input.txt", 'r') as f:
    lines = f.readlines()

# part 1
def find_two_numbers(lines):
    n = len(lines)
    for i in range(n):
        for j in range(i+1, n):
            if int(lines[i]) + int(lines[j]) == 2020:
                return int(lines[i]) * int(lines[j])

# part 2
def find_three_numbers(lines):
    n = len(lines)
    for i in range(n):
        for j in range(i+1, n):
            for k in range(j+1, n):
                if int(lines[i]) + int(lines[j]) + int(lines[k]) == 2020:
                    return int(lines[i]) * int(lines[j]) * int(lines[k])

File: 2020/day01/java/ReportRepair.java

import java.nio.file.Files;
import java.nio.file.Path;
import java.util.ArrayList;
import java.util.List;

public class ReportRepair {

	public static void main(String[] args) {
		readLines("input.txt");
	}

	public static void readLines(String file) {
		List<String> lines = new ArrayList<>();
		try {
			lines = Files.readAllLines(Path.of(file));
		} catch (Exception e) {
			System.out.println("Error: " + e.getMessage());
		}

		int part1 = findTwoNumbers(lines);
		int part2 = findThreeNumbers(lines);
		System.out.println(part1);
		System.out.println(part2);

	}

	public static int findThreeNumbers(List<String> numbers){
		for(int i = 0; i < numbers.size(); i++){
			for(int j = 0; j < i; j++){
				for(int k = 0; k < j; k++){
					int num1 = Integer.parseInt(numbers.get(i));
					int num2 = Integer.parseInt(numbers.get(j));
					int num3 = Integer.parseInt(numbers.get(k));
					if(num1 + num2 + num3 == 2020){
						return num1 * num2 * num3;
					}
				}
			}
		}
		return -1;
	}

	public static int findTwoNumbers(List<String> numbers) {
		for(int i = 0; i < numbers.size(); i++){
			for(int j = i+1; j < numbers.size(); j++){
				int num1 = Integer.parseInt(numbers.get(i));
				int num2 = Integer.parseInt(numbers.get(j));
				if(num1 + num2 == 2020){
					return num1 * num2;
				}
			}
		}
		return -1;
	}
}
